spec/fenrir_view/property_types_spec.rb

```diff
@@ -0,0 +1,299 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe FenrirView::PropertyTypes do
+  let(:mock_instance) {
+    lambda { |instance_props, component_props|
+      FenrirView::PropertyTypes.new({
+                                      component_class: String,
+                                      component_properties: component_props,
+                                      instance_properties: instance_props,
+                                    })
+    }
+  }
+
+  let(:mock_component) {
+    lambda { |content, validations|
+      FenrirView::PropertyTypes.new({
+                                      component_class: String,
+                                      component_properties: {},
+                                      instance_properties: {},
+                                    }).validate_property(:title, content, validations)
+    }
+  }
+
+  let(:title_property) { 'The best title' }
+
+  describe 'Presenter initializes validator' do
+    let(:card_facade_property_types) {
+      FenrirView::Presenter.component_for('component', 'card', {
+        title: 'everything',
+      }, validate: true).send(:property_types)
+    }
+
+    let(:component_properties) do
+      {
+        yield: {},
+        title: { required: true },
+        description: {},
+        link: {},
+        image_url: {},
+        location: {},
+        data: { default: [] }
+      }
+    end
+
+    let(:instance_properties) do
+      {
+        yield: nil,
+        title: 'everything',
+        description: nil,
+        link: nil,
+        image_url: nil,
+        location: nil,
+        data: []
+      }
+    end
+
+    it 'Presenter can initialize validator' do
+      expect(card_facade_property_types).to respond_to(:validate_properties)
+
+      expect(card_facade_property_types.instance_variable_get(:@component_class)).to eq(CardFacade)
+      expect(card_facade_property_types.instance_variable_get(:@component_properties)).to eq(component_properties)
+      expect(card_facade_property_types.instance_variable_get(:@instance_properties)).to eq(instance_properties)
+    end
+  end
+
+  describe '#validate_properties' do
+    it 'returns instance properties when correct' do
```

Comment:
Currently this method returns things, maybe it should return `:ok` or something instead?

Reply:
Good shout, I might leave these changes to a separate PR though. 

Currently I've decided to not validate on production to make sure these components render quicker. But maybe we'd want any invalid properties on production to show up in the logs?